src/email.rs

```diff
@@ -0,0 +1,169 @@
+use std::ffi::CString;
+use std::{mem, ptr};
+
+use crate::util::Binding;
+use crate::{raw, Buf, Commit, DiffFindOptions, DiffOptions, Error, IntoCString};
+use crate::{Diff, Oid, Signature};
+
+/// A structure to represent patch in mbox format for sending via email
+pub struct Email {
+    /// Buffer to store the e-mail patch in
+    pub buf: Buf,
```

Comment:
Instead of making this a `pub` field could public accessors be provided for this? (sorry missed this earlier)

Ideally byte slices would be returned as well instead of `&Buf` too

Reply:
Done.